pages/search/index.js

```diff
@@ -0,0 +1,41 @@
+import SearchCard from "@/components/Search/SearchCard";
+import EventList from "@/components/EventList/EventList";
+import { useState } from "react";
+import useSWR from "swr";
+import Loading from "@/components/Loading/Loading";
+import { SearchMessage } from "@/components/Search/Search.styled";
+
+export default function SearchPage() {
+  const [searchTerm, setSearchTerm] = useState("");
+  const { data: filteredEvents, isLoading } = useSWR(
+    searchTerm ? `/api/search/${searchTerm}` : null
+  );
+  const hasSearchResults = filteredEvents
+    ? filteredEvents.length > 0
+      ? true
+      : false
+    : undefined;
```

Comment:
```suggestion
 const hasSearchResults = Boolean(filteredEvents?.length);
```
You could shorten this with the Boolean constructor
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean/Boolean

Reply:
But... 
I need it to be undefined when !filteredEvents, because then user wont see "Keine Suchergebnisse" before entering a search term the first time. 

so this is a way: 
  const hasSearchResults = filteredEvents
    ? Boolean(filteredEvents.length)
    : undefined;